static/figure/js/models/figure_model.js

```diff
@@ -45,47 +45,53 @@
 
             var load_url = BASE_WEBFIGURE_URL + "load_web_figure/" + fileId + "/",
                 self = this;
-
-
+            
             $.getJSON(load_url, function(data){
-
-                // bring older files up-to-date
-                data = self.version_transform(data);
-
-                var name = data.figureName || "UN-NAMED",
-                    n = {'fileId': fileId,
-                        'figureName': name,
-                        'canEdit': data.canEdit,
-                        'paper_width': data.paper_width,
-                        'paper_height': data.paper_height,
-                        'page_size': data.page_size || 'letter',
-                        'page_count': data.page_count,
-                        'paper_spacing': data.paper_spacing,
-                        'page_col_count': data.page_col_count,
-                        'orientation': data.orientation,
-                        'legend': data.legend,
-                        'legend_collapsed': data.legend_collapsed,
-                    };
-
-                // For missing attributes, we fill in with defaults
-                // so as to clear everything from previous figure.
-                n = $.extend({}, self.defaults, n);
-
-                self.set(n);
-
-                _.each(data.panels, function(p){
-                    p.selected = false;
-                    self.panels.create(p);
-                });
-
-                self.set('unsaved', false);
-                // wait for undo/redo to handle above, then...
-                setTimeout(function() {
-                    self.trigger("reset_undo_redo");
-                }, 50);
+              data.fileId = fileId;
+              self.load_from_JSON(data);
             });
+            
+            self.set('unsaved', false);
```

Comment:
This line needs to be called within the ```$.getJSON`` callback above, right after ``self.load_from_JSON(data)```. Currently this is getting called before the data loads.